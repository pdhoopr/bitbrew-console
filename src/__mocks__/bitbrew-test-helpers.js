import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import diff from 'jest-diff';
import { addSerializer, getSerializers, utils } from 'jest-snapshot';
import styledSerializer from 'jest-styled-components/src/styleSheetSerializer';
import stripAnsi from 'strip-ansi';

const DIFF_HEADER = 'Diff:\n';

export const diffComponents = (baseComponent, compareComponent) => {
  const serializers = getSerializers();
  [styledSerializer, enzymeSerializer].forEach(serializer => {
    if (!serializers.includes(serializer)) {
      addSerializer(serializer);
    }
  });
  const base = `${utils.serialize(shallow(baseComponent))}\n`;
  const compare = `${utils.serialize(shallow(compareComponent))}\n`;
  const changes = diff(base, compare, {
    aAnnotation: `<BaseComponent />`,
    bAnnotation: `<CompareComponent />`,
    contextLines: 3,
    expand: false,
  });
  return `${DIFF_HEADER}${changes}`;
};

export const diffSerializer = {
  test(value) {
    return typeof value === 'string' && value.startsWith(DIFF_HEADER);
  },
  print(value) {
    return stripAnsi(value).trim();
  },
};

export const fixtures = {
  accessToken:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE1MjUxMTUzNTAsImV4cCI6MTU1NjY1MTM1MCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSJ9.4O-W8aeOOUFU8fkGTAFaAu9D9nK_l61yPomNG1Qkny8',
  orgs: {
    danlaw: {
      id: 'bd006616-3bd2-4783-a84d-b0fa4ed81135',
      properName: 'Danlaw',
    },
    azuga: {
      id: '8d9ec637-f3bb-4c46-83ee-11764d09f0b2',
      properName: 'Azuga',
    },
    bitbrew: {
      id: '94d9b0ca-a758-4502-8dcd-41d883e38714',
      properName: 'BitBrew',
    },
  },
  users: {
    bill: {
      email: 'bill@bitbrew.com',
      name: 'Bill Monaghan',
    },
    cynthia: {
      email: 'cynthia@bitbrew.com',
      name: 'Cynthia Hamilton',
    },
    alex: {
      email: 'bill@bitbrew.com',
      name: 'Alex Li',
    },
  },
};
