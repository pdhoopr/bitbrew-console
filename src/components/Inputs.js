import styled, { css } from 'styled-components';

export const Label = styled.label`
  display: block;
  font-weight: var(--weight-bold);
  margin-bottom: var(--size-16);
`;

export const baseStyles = css`
  border-radius: var(--corner-radius);
  color: var(--color-black);
  display: block;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  line-height: var(--size-20);
  width: 100%;

  &::placeholder {
    color: var(--color-dark-grey);
  }
`;

export const Input = styled.input`
  ${baseStyles};
  border: 1px solid var(--color-medium-grey);
  padding: var(--size-7) var(--size-16);

  ${/* sc-selector */ Label} & {
    margin-top: var(--size-8);
  }
`;

export const RaisedInput = styled.input`
  ${baseStyles};
  border: none;
  box-shadow: var(--elevation-low);
  padding: var(--size-8) var(--size-16);
`;
