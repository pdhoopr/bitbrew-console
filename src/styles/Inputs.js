import styled, { css } from 'styled-components';

const baseStyles = css`
  border-radius: var(--corner-radius);
  color: var(--color-black);
  display: block;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  line-height: var(--size-20);
  width: 100%;

  &::placeholder {
    color: var(--color-dark-grey);
  }
`;

export const Input = styled.input`
  ${baseStyles};
  border: 1px solid var(--color-medium-grey);
  padding: calc(var(--size-8) - 1px) var(--size-16);
`;

export const RaisedInput = styled.input`
  ${baseStyles};
  border: none;
  box-shadow: var(--elevation-low);
  padding: var(--size-8) var(--size-16);
`;
