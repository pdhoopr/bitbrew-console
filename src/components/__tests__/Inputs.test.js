import { diffComponents } from 'bitbrew-test-helpers';
import { shallow } from 'enzyme';
import React from 'react';
import styled from 'styled-components';
import { baseStyles, Input, RaisedInput } from '../Inputs';

const BaseInput = styled.input`
  ${baseStyles};
`;

test('renders DOM shared by all inputs', () => {
  const dom = shallow(<BaseInput />);
  expect(dom).toMatchSnapshot();
});

test('renders DOM specific to a default input', () => {
  const base = <BaseInput />;
  const comp = <Input />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});

test('renders DOM specific to a raised input', () => {
  const base = <BaseInput />;
  const comp = <RaisedInput />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});
