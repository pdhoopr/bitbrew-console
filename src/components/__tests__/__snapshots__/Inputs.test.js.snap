// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM shared by all inputs 1`] = `
.c0 {
  border-radius: var(--corner-radius);
  color: var(--color-black);
  display: block;
  font-family: var(--font-roboto);
  font-size: var(--size-14);
  line-height: var(--size-20);
  width: 100%;
}

.c0::-webkit-input-placeholder {
  color: var(--color-dark-grey);
}

.c0::-moz-placeholder {
  color: var(--color-dark-grey);
}

.c0:-ms-input-placeholder {
  color: var(--color-dark-grey);
}

.c0::placeholder {
  color: var(--color-dark-grey);
}

<input
  className="c0"
/>
`;

exports[`renders DOM specific to a default input 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -7,6 +7,8 @@
    font-size: var(--size-14);
    line-height: var(--size-20);
    width: 100%;
+   border: 1px solid var(--color-medium-grey);
+   padding: calc(var(--size-8) - 1px) var(--size-16);
  }
  
  .c0::-webkit-input-placeholder {
`;

exports[`renders DOM specific to a raised input 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -7,6 +7,9 @@
    font-size: var(--size-14);
    line-height: var(--size-20);
    width: 100%;
+   border: none;
+   box-shadow: var(--elevation-low);
+   padding: var(--size-8) var(--size-16);
  }
  
  .c0::-webkit-input-placeholder {
`;
